### ISSUE TITLE
Connecting to postgres metabase db over ssl

### ISSUE THREAD

**Issue by blakeblackshear** [metabase/metabase#3296]
- Chrome
- Linux (CentOS 7)
- JAR
- Internal database is Postgres
- Data is in redshift
- Version 0.19.3

I have been unable to figure out how to connect to the internal database over SSL. My environment variables are set to:
```
MB_DB_TYPE=postgres
MB_DB_HOST=postgres.service.consul
MB_DB_NAME=metabase
MB_DB_PORT=5432
MB_DB_USER=metabase
MB_DB_PASS=password
```

I get the following on startup:
```
08-31 22:50:55 #033[1mERROR metabase.driver#033[0m :: Failed to connect to database: org.postgresql.util.PSQLException: FATAL: no pg_hba.conf entry for host "10.0.157.186", user "metabase", database "metabase", SSL off
java.lang.AssertionError: Assert failed: Unable to connect to Metabase postgres DB.
```

It looks like I might be able to tweak the `MB_DB_CONNECTION_URI` environment variable to get this to work, but I am not sure how.

This instance was previously connected over a non-ssl connection.

**Comment by camsaul** [comment]
Hi @blakeblackshear use`METABASE_DB_CONNECTION_URI` if you want use SSL to connect to the internal database. The string should use the following format:

```
postgres://<username>:<password>@<host>:<port>/<dbname>?ssl=true&sslfactory=org.postgresql.ssl.NonValidatingFactory
```

**Comment by blakeblackshear** [comment]
That worked. Thanks.

Went ahead and opened a PR to add this to the docs: https://github.com/metabase/metabase/pull/3299

**Comment by sameersoni** [comment]
Hi,
I am using Metabase version v0.26.2.
Where is the option for mentioning `METABASE_DB_CONNECTION_URI` variable directly?

While trying to create Database connection via Metabase ADD DATABASE page, I tried giving options `ssl=true&sslfactory=org.postgresql.ssl.NonValidatingFactory` under Additional JSBC options, but it didn't work for me.

Please suggest.

**Comment by erkie** [comment]
For future spelunkers: It's `MB_DB_CONNECTION_URI` not `METABASE_DB_CONNECTION_URI` (found this thread via Google search)

**Comment by KazW** [comment]
Just adding to this thread for people Googling this when encountering this error. If you happen to be hosting Metabase on Heroku and are moving your application DB from Heroku to RDS, **don't add parameters to the** `DATABASE_URL`. The Metabase buildpack adds them for you, so your `DATABASE_URL` should end with the name of the database and SSL will be configured automatically.

**Comment by normkatz** [comment]
I tried argument MB_DB_CONNECTION_URI=postgresql://myuser:mypassword@mydb-host:port/metabase?ssl=true&sslfactory=org.postgresql.ssl.NonValidatingFactory

I got this error:
org.postgresql.util.PSQLException: Could not open SSL root certificate file /home/metabase/.postgresql/root.crt.

I'm running postgres 10.  I thought ssl=true is deprecated and replaced with sslmode=require (or other choice).  But my driver did try SSL anyway.  It just couldn't find a cert in my container.  Are there instructions for creating one in metabase docs or should it already be there?
